app/src/main/java/com/hedvig/app/feature/insurance/ui/InsuranceAdapter.kt

```diff
@@ -219,24 +298,133 @@ class InsuranceAdapter(
 
     companion object {
         private const val TRANSITION_NAME = "contract_card"
+
+        private const val CROSS_SELL = 1
+        private const val SUBHEADING = 2
     }
 }
 
-sealed class InsuranceModel {
-    object Header : InsuranceModel()
+@Composable
+fun Subheading(text: String) {
+    Text(
+        text = text,
+        style = MaterialTheme.typography.h6,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(
+                start = 16.dp,
+                top = 48.dp,
+                end = 16.dp,
+                bottom = 8.dp,
+            ),
+    )
+}
 
-    data class Contract(
-        val inner: InsuranceQuery.Contract
-    ) : InsuranceModel()
+@Preview(showBackground = true)
+@Composable
+fun SubheadingPreview() {
+    HedvigTheme {
+        Subheading("Add more coverage")
+    }
+}
 
-    data class Upsell(
-        @get:StringRes val title: Int,
-        @get:StringRes val description: Int,
-        @get:StringRes val ctaText: Int
-    ) : InsuranceModel()
+/*
+ * Note: This Composable uses hardcoded colors due to difficulties with
+ * declaring a particular component to be in dark theme instead of the
+ * default. When we update `HedvigTheme` to be Compose-first instead of
+ * XML-Theme first, we can reconfigure the theme for this composable to
+ * be `dark` no matter what the system value is.
+ */
+@Composable
+fun CrossSell(
+    data: InsuranceModel.CrossSell,
```

Comment:
I think we can make this reusable by passing the primitive types instead of `InsuranceModel.CrossSell` - like `backgroundBlurHash`, `backgroundUrl` etc.

Reply:
I'd agree if we had more use-cases for this particular type of card - for now, I'd point to the rule of three 😉 

https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
Fair enough :)